content/fr/articles/js/react/styled-components/index.md

```diff
@@ -0,0 +1,308 @@
+---
+date: "2017-01-31"
+title: "Introduction à styled-components"
+tags:
+  - javascript
+  - react
+  - css
+authors:
+  - revolunet
+
+---
+
+Il existe de nombreuses d'approches et solutions différentes pour gérer le CSS en ReactJS, je vais parler ici de [styled-components](https://github.com/styled-components/styled-components) qui permet de gagner en productivité et en qualité quand on veut créer et distribuer des composants UI.
+
+La force de [styled-components](https://github.com/styled-components/styled-components) c'est de faciliter la création de composants visuels ReactJS minimalistes et configurables, en combinant du **CSS standard** et **un zeste de JavaScript**. Ces composants deviendront littéralement les pièces de lego qui pourront être utilisée et partagées entre les UIs de toutes vos applications, sites webs...
+
+Si vous voulez jouer avec, je vous conseille d'utiliser [create-react-app](https://github.com/facebookincubator/create-react-app) pour générer un projet vierge puis d'ajouter styled-components au projet : `npm i --save styled-components`.
+
+Avec styled, on peut abstraire le CSS et le DOM de la vue, ce qui va grandement simplifier le code de la partie "métier" de l'application, et améliorer la lisibilité.
+
+##### Avant :
+
+```jsx
+<div className={ styles.box }>
+  <div className={ styles.title }>{ title }</div>
+  <div className={ classnames(styles.description, styled.important) }>{ description }</div>
+</div>
+```
+
+##### Après :
+
+```jsx
+<Box>
+  <Title>{ title }</Title>
+  <Description important>{ description }</Description>
+</Box>
+```
+
+### Avantages :
+ - [community powered](https://github.com/styled-components/styled-components/graphs/contributors) - moving fast, [v2 en cours](https://github.com/styled-components/styled-components/tree/v2)
+ - support des [thèmes](https://github.com/styled-components/styled-components/blob/master/docs/theming.md)
+ - [syntaxe CSS "classique"](https://github.com/styled-components/styled-components/blob/master/docs/css-we-support.md) mais scopé automatiquement
+ - Gestion des media queries, pseudo-selectors, nesting
+ - Autroprefixer intégré
+ - Code "métier" plus lisible
```

Comment:
ce que je veux dire c'est que le "bruit" généré par le styling ne pollue pas le code de l'application elle-même mais sont isolés. (separation of concerns?). suggestion ?